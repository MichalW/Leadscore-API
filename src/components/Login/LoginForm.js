import React, {PureComponent} from 'react';
import {
  Button, Form, Input, Label, Segment,
} from 'semantic-ui-react';
import PropTypes from 'prop-types';

class LoginForm extends PureComponent {
  state = {
    username: '',
    password: '',
  };

  onChangeUsername = (event) => {
    this.setState({username: event.target.value});
  };

  onChangePassword = (event) => {
    this.setState({password: event.target.value});
  };

  onSubmit = () => {
    const {signInRequest} = this.props;
    const {username, password} = this.state;

    signInRequest(username, password);
  };

  render() {
    const {username, password} = this.state;

    return (
      <Form onSubmit={this.onSubmit}>
        <Segment>
          <Form.Field>
            <Label>Login</Label>
            <Input placeholder="Login" onChange={this.onChangeUsername} value={username} />
          </Form.Field>

          <Form.Field>
            <Label>Password</Label>
            <input placeholder="Password" type="password" onChange={this.onChangePassword} value={password} />
          </Form.Field>

          <Button type="submit" fluid primary>Login</Button>
        </Segment>
      </Form>
    );
  }
}

LoginForm.propTypes = {
  signInRequest: PropTypes.func.isRequired,
};

export default LoginForm;
