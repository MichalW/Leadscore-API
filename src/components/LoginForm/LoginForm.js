import React, {PureComponent} from 'react';
import {
  Button, Form, Input, Label, Segment, Message,
} from 'semantic-ui-react';
import PropTypes from 'prop-types';

class LoginForm extends PureComponent {
  state = {
    username: '',
    password: '',
  };

  onChangeUsername = ({target: {value}}) => {
    this.setState({username: value});
  };

  onChangePassword = ({target: {value}}) => {
    this.setState({password: value});
  };

  onSubmit = () => {
    const {signInRequest} = this.props;
    const {username, password} = this.state;

    signInRequest(username, password);
  };

  render() {
    const {error, loading} = this.props;
    const {username, password} = this.state;

    return (
      <Form onSubmit={this.onSubmit}>
        <Segment>
          <Form.Field>
            <Label>Login</Label>
            <Input placeholder="Login" onChange={this.onChangeUsername} value={username} />
          </Form.Field>

          <Form.Field>
            <Label>Password</Label>
            <input placeholder="Password" type="password" onChange={this.onChangePassword} value={password} />
          </Form.Field>

          {error && (
            <Message negative>
              <p>Invalid credentials</p>
            </Message>
          )}

          <Button type="submit" fluid primary loading={loading}>Login</Button>
        </Segment>
      </Form>
    );
  }
}

LoginForm.propTypes = {
  error: PropTypes.bool.isRequired,
  loading: PropTypes.bool.isRequired,
  signInRequest: PropTypes.func.isRequired,
};

export default LoginForm;
