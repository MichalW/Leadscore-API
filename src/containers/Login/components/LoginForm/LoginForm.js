import React, {PureComponent} from 'react';
import {
  Button, Form, Input, Message, Segment,
} from 'semantic-ui-react';
import PropTypes from 'prop-types';

class LoginForm extends PureComponent {
  state = {
    username: '',
    password: '',
  };

  onChangeUsername = ({target: {value}}) => {
    this.setState({username: value});
  };

  onChangePassword = ({target: {value}}) => {
    this.setState({password: value});
  };

  onSubmit = () => {
    const {signInRequest} = this.props;
    const {username, password} = this.state;

    signInRequest(username, password);
  };

  render() {
    const {error, loading} = this.props;
    const {username, password} = this.state;

    return (
      <Form onSubmit={this.onSubmit}>
        <Segment>
          <Form.Field
            control={Input}
            label="Login"
            onChange={this.onChangeUsername}
            placeholder="Login"
            required
            value={username}
          />

          <Form.Field
            control={Input}
            label="Password"
            onChange={this.onChangePassword}
            placeholder="Password"
            required
            type="password"
            value={password}
          />

          {error && (
            <Message negative>
              <p>Invalid credentials</p>
            </Message>
          )}

          <Button type="submit" fluid primary loading={loading}>Login</Button>
        </Segment>
      </Form>
    );
  }
}

LoginForm.propTypes = {
  error: PropTypes.bool.isRequired,
  loading: PropTypes.bool.isRequired,
  signInRequest: PropTypes.func.isRequired,
};

export default LoginForm;
