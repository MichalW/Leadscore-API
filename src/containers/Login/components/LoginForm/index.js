import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';

import LoginForm from './LoginForm';
import {signInRequest} from '../../../../actions/authentication';

const mapStateToProps = ({authentication: {error, loading}}) => ({
  error,
  loading,
});

const mapDispatchToProps = dispatch => ({
  signInRequest: bindActionCreators(signInRequest, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(LoginForm);
