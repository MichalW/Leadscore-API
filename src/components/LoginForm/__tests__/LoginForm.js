import React from 'react';
import renderer from 'react-test-renderer';

import LoginForm from '../LoginForm';

it('renders LoginForm without crashing', () => {
  const tree = renderer
    .create(<LoginForm signInRequest={() => {}} loading={false} error={false} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders LoginForm with error', () => {
  const tree = renderer
    .create(<LoginForm signInRequest={() => {}} loading={false} error />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders LoginForm with loading', () => {
  const tree = renderer
    .create(<LoginForm signInRequest={() => {}} loading error={false} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
