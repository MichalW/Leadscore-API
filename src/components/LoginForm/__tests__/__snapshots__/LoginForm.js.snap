// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LoginForm with error 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="ui segment"
  >
    <div
      className="field"
    >
      <div
        className="ui label"
        onClick={[Function]}
      >
        Login
      </div>
      <div
        className="ui input"
      >
        <input
          onChange={[Function]}
          placeholder="Login"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="ui label"
        onClick={[Function]}
      >
        Password
      </div>
      <input
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
    <div
      className="ui negative message"
    >
      <p>
        Invalid credentials
      </p>
    </div>
    <button
      className="ui fluid primary button"
      onClick={[Function]}
      type="submit"
    >
      Login
    </button>
  </div>
</form>
`;

exports[`renders LoginForm with loading 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="ui segment"
  >
    <div
      className="field"
    >
      <div
        className="ui label"
        onClick={[Function]}
      >
        Login
      </div>
      <div
        className="ui input"
      >
        <input
          onChange={[Function]}
          placeholder="Login"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="ui label"
        onClick={[Function]}
      >
        Password
      </div>
      <input
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
    <button
      className="ui fluid loading primary button"
      onClick={[Function]}
      type="submit"
    >
      Login
    </button>
  </div>
</form>
`;

exports[`renders LoginForm without crashing 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="ui segment"
  >
    <div
      className="field"
    >
      <div
        className="ui label"
        onClick={[Function]}
      >
        Login
      </div>
      <div
        className="ui input"
      >
        <input
          onChange={[Function]}
          placeholder="Login"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="ui label"
        onClick={[Function]}
      >
        Password
      </div>
      <input
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
    <button
      className="ui fluid primary button"
      onClick={[Function]}
      type="submit"
    >
      Login
    </button>
  </div>
</form>
`;
